src/clarity-angular/popover/signpost/signpost.specs.ts

```diff
@@ -0,0 +1,485 @@
+             * 1. Correct class name
+             * 2. correct anchor point
+             * 3. correct popover point
+             * 4. Correct Y offset
+             * 5. Correct X offset
+             *
+             */
+
+            signpost.signpostPosition = "left-top";
+            fixture.detectChanges();
+            let testClass = clarityElement.querySelector(".left-top");
+            expect(testClass).toBeDefined();
+
+            expect(signpost.anchorPoint).toBe(Point.LEFT_TOP);
+            expect(signpost.popoverPoint).toBe(Point.RIGHT_TOP);
+            expect(signpost.signpostOptions.offsetY).toBe(18);
+            expect(signpost.signpostOptions.offsetX).toBe(-14);
+        });
+    });
+
+    describe("SignpostContent", function() {
+        let fixture: ComponentFixture<any>;
+        let clarityElement: any;
+        let signpostContent: SignpostContent;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                imports: [ ClrSignpostModule, ClrIconModule ],
+                declarations: [ TestSignpost ]
+            });
+
+            fixture = TestBed.createComponent(TestSignpost);
+            fixture.detectChanges();
+            clarityElement = fixture.nativeElement;
+            signpostContent = fixture.componentInstance.content;
+        });
+
+        afterEach(() => {
+            fixture.destroy();
+        });
+
+        it("projects content when open", function() {
+            let signpostToggle: HTMLElement = clarityElement.querySelector(".btn");
+            let projectedContent: HTMLElement;
+
+            signpostToggle.click();
+            fixture.detectChanges();
+            projectedContent = clarityElement.querySelector(".signpost-content-body").textContent.trim();
+            expect(projectedContent).toEqual("Signpost content");
+        });
+
+        it("has a close button", function() {
+            let signpostToggle: HTMLElement = clarityElement.querySelector(".btn");
+            let content: HTMLElement;
+
+            signpostToggle.click();
+            fixture.detectChanges();
+            let closer: HTMLElement = clarityElement.querySelector(
+                ".signpost-content > .signpost-flex-wrap > .signpost-content-header > .close"
+            );
+            expect(closer).toBeDefined();
+            closer.click();
+
+            fixture.detectChanges();
+
+            content = clarityElement.querySelector(".signpost-content");
+            expect(content).toBeNull();
+        });
+
+        it("closes the signpost when clicked outside host", function() {
+            let signpostToggle: HTMLElement = clarityElement.querySelector(".btn");
+            signpostToggle.click();
+            fixture.detectChanges();
+            expect(clarityElement.querySelector(".signpost-content")).not.toBeNull();
+
+            let outsideButton: HTMLElement = clarityElement.querySelector(".outside-click-test");
```

Comment:
You probably want to test that the signpost doesn't close when its own content is clicked.